lib/io/can.cpp

```diff
@@ -1,8 +1,79 @@
 #include "can.hpp"
 
+#include <stdio.h>
+#include <unistd.h>
+
+#include <linux/can.h>
+#include <net/if.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+
 namespace hyped::io {
-Can::Can()
+Can::Can(hyped::core::ILogger &logger) : logger_(logger)
+{
+}
+
+CanResult Can::initialiseCanSocket(std::string can_network_interface)
```

Comment:
Just `initialise`?

Re semantics: We don't accommodate for the possibility of being on more than one CAN network, right?

Reply:
I'm pretty sure, but haven't tested, that this all works with multiple networks if we instantiate a new Can class and initialise with a new network interface. But also we shouldn't ever have more than one bus, containing BMS, controller, and (maybe) nucleo board